service/adapters/bolt/social_graph_repository_test.go

```diff
@@ -74,3 +73,94 @@ func TestSocialGraphRepository_RemoveDropsContactDataForTheSpecifiedFeed(t *test
 	})
 	require.NoError(t, err)
 }
+
+func TestSocialGraphRepository_GetContacts(t *testing.T) {
+	db := fixtures.Bolt(t)
+
+	iden := fixtures.SomeRefIdentity()
+
+	target1 := fixtures.SomeRefIdentity()
+	target2 := fixtures.SomeRefIdentity()
+	target3 := fixtures.SomeRefIdentity()
+
+	err := db.Update(func(tx *bbolt.Tx) error {
+		txadapters, err := di.BuildTxTestAdapters(tx)
+		require.NoError(t, err)
+
+		applyContactAction(t, txadapters, iden, target1, content.ContactActionFollow)
+		applyContactAction(t, txadapters, iden, target2, content.ContactActionFollow)
+		applyContactAction(t, txadapters, iden, target3, content.ContactActionFollow)
+
+		return nil
+	})
+	require.NoError(t, err)
+
+	err = db.View(func(tx *bbolt.Tx) error {
+		txadapters, err := di.BuildTxTestAdapters(tx)
+		require.NoError(t, err)
+
+		contacts, err := txadapters.SocialGraphRepository.GetContacts(iden)
+		require.NoError(t, err)
+		sortAndCompareContacts(t,
+			[]*feeds.Contact{
+				feeds.MustNewContactFromHistory(target1, true, false),
+				feeds.MustNewContactFromHistory(target2, true, false),
+				feeds.MustNewContactFromHistory(target3, true, false),
+			},
+			contacts,
+		)
+
+		return nil
+	})
+	require.NoError(t, err)
+
+	err = db.Update(func(tx *bbolt.Tx) error {
+		txadapters, err := di.BuildTxTestAdapters(tx)
+		require.NoError(t, err)
+
+		applyContactAction(t, txadapters, iden, target1, content.ContactActionBlock)
+		applyContactAction(t, txadapters, iden, target2, content.ContactActionUnfollow)
+
+		return nil
+	})
+	require.NoError(t, err)
+
+	err = db.View(func(tx *bbolt.Tx) error {
+		txadapters, err := di.BuildTxTestAdapters(tx)
+		require.NoError(t, err)
+
+		contacts, err := txadapters.SocialGraphRepository.GetContacts(iden)
+		require.NoError(t, err)
+		sortAndCompareContacts(
+			t,
+			[]*feeds.Contact{
+				feeds.MustNewContactFromHistory(target1, true, true),
+				feeds.MustNewContactFromHistory(target2, false, false),
+				feeds.MustNewContactFromHistory(target3, true, false),
+			},
+			contacts,
+		)
+
+		return nil
+	})
+	require.NoError(t, err)
+}
+
+func sortAndCompareContacts(t *testing.T, a []*feeds.Contact, b []*feeds.Contact) {
```

Comment:
Maybe `sortAndRequireEqual` would be a more suitable name here?

Reply:
cb4122f